galley/pkg/config/analysis/analyzers/util/hosts.go

```diff
@@ -47,3 +48,19 @@ func getNamespaceAndNameFromFQDN(fqdn string) (string, string) {
 	}
 	return result[0][2], result[0][1]
 }
+
+// GetScopedFqdnHostname converts the passed host to FQDN if needed and applies the passed scope.
+func GetScopedFqdnHostname(scope, namespace, host string) string {
```

Comment:
Should we give "scoped FQDN" its own named type? e.g. 
```type scopedFqdn string``` 

I think it might make semantics easier to follow and even give us a little bit of compile-time type checking that we're passing around what we think we are.

Reply:
yep, good idea. Will do.